Lib/glyphsLib/builder/color_layers.py

```diff
@@ -0,0 +1,151 @@
+    for (glyph, masterLayer), layers in self._color_palette_layers:
+        ufo_font = self._sources[
+            masterLayer.associatedMasterId or masterLayer.layerId
+        ].font
+        colorLayers = []
+        for i, (layer, colorId) in enumerate(layers):
+            if layer != masterLayer:
+                layerGlyphName = f"{glyph.name}.color{i}"
+                ufo_layer = self.to_ufo_layer(glyph, masterLayer)
+                ufo_glyph = ufo_layer.newGlyph(layerGlyphName)
+                self.to_ufo_glyph(ufo_glyph, layer, glyph)
+            else:
+                layerGlyphName = glyph.name
+            colorLayers.append((layerGlyphName, colorId))
+        ufo_font.lib.setdefault(UFO2FT_COLOR_LAYERS_KEY, {})[glyph.name] = colorLayers
+
+    palette = []
+    for (glyph, masterLayer), layers in self._color_layers:
+        masterId = masterLayer.associatedMasterId or masterLayer.layerId
+        ufo_font = self._sources[masterId].font
+
+        colorLayers = ufo_font.lib.setdefault(UFO2FT_COLOR_LAYERS_KEY, {})
+        assert glyph.name not in colorLayers
+
+        paintLayers = []
+        for i, layer in enumerate(layers):
+            if layer != masterLayer:
+                layerGlyphName = f"{glyph.name}.color{i}"
+                ufo_layer = self.to_ufo_layer(glyph, masterLayer)
+                ufo_glyph = ufo_layer.newGlyph(layerGlyphName)
+                self.to_ufo_glyph(ufo_glyph, layer, glyph)
+            else:
+                layerGlyphName = glyph.name
+
+            attributes = layer.paths[0].attributes
+            if "gradient" in attributes:
+                gradient = attributes["gradient"]
+
+                # Glyphs start and stop points seem to be a percentage of the
+                # path size, but we want absolute coordinates.
+                bounds = layer.bounds
+                x0, y0 = gradient["start"]
+                x1, y1 = gradient["end"]
+                x0 = bounds.origin.x + (bounds.size.width * x0)
+                y0 = bounds.origin.y + (bounds.size.height * y0)
+                x1 = bounds.origin.x + (bounds.size.width * x1)
+                y1 = bounds.origin.y + (bounds.size.height * y1)
+
+                colorStop = []
+                for color in gradient.get("colors"):
+                    (r, g, b, a), o = color
+                    paletteIndex, a = _find_or_insert_color((r, g, b, a), palette)
+                    colorStop.append(
+                        dict(StopOffset=o, Alpha=a, PaletteIndex=paletteIndex)
+                    )
+
+                colorLine = dict(Extend="pad", ColorStop=colorStop)
+
+                gradientType = gradient.get("type")
+                if gradientType is None:
+                    # Linear gradient.
+                    # We set point 2 to point 1 rotated -90 degrees around
+                    # point 0, since Glyphs gradients don’t represent them.
+                    paint = dict(
+                        Format=4,
```

Comment:
I suggest importing the PaintFormat enum from fontTools otTables module instead of hard coding the format number.